### DISCUSSION TITLE
Library uses vue-demi with vue3, app uses vue2 but still vue-demi detects vue3

### DISCUSSION THREAD

**Discussion by pareshchouhan** [vueuse/vue-demi#122] [Q&A]
I am using vue-cli-service to compile my library which uses vue-demi and I develop the code using vue 3.2.X.
When I import the library in a vue2 project `vue-demi` detects vue3 is running instead of vue2.

This is a console log of following in my library, which I imported in my application.
```js
import * as VueDemi from 'vue-demi';
console.log(VueDemi)
```

![image](https://user-images.githubusercontent.com/1715271/144395139-b0518c23-a81a-4ba6-a1af-7e90486cf100.png)

Here's a repo to reproduce it 
[https://github.com/pareshchouhan/vue-demi-test](https://github.com/pareshchouhan/vue-demi-test)  

In my library only thing I use vue-demi is for importing defineComponent

```js
import { defineComponent } from 'vue-demi';
````

It crashes inside the library code when calling `defineComponent`
![image](https://user-images.githubusercontent.com/1715271/144395874-870a2ec7-bca5-4419-a0e7-4fb271111eac.png)

**Comment by antfu** [comment]
You need to externalize `vue-demi`

**Comment by pareshchouhan** [comment]
@antfu , it is externalised, I checked the source code in the built files.

```
// EXTERNAL MODULE: D:/cumul.io/cumulio-libs/node_modules/vue-demi/lib/index.mjs
var lib = __webpack_require__("612a");

// EXTERNAL MODULE: external {"commonjs":"vue","commonjs2":"vue","root":"Vue"}
var external_commonjs_vue_commonjs2_vue_root_Vue_ = __webpack_require__("8bbf");
```

**Reply by rookieLink** [reply]
You need to externalize "vue-demi"， not only "vue"，when "vue-demi" install in you project, it will process postInstall to recognition your vue version in your project

**Comment by pareshchouhan** [comment]
I added it to webpack configuration as well under externals, and still same results :)
